node/src/signature_provider/ecdsa_secp256k1/mod.rs

```diff
@@ -0,0 +1,228 @@
+use crate::network::{MeshNetworkClient, NetworkTaskChannel};
+use crate::primitives::{HasParticipants, MpcTaskId};
+use crate::sign_request::{SignRequestStorage, SignatureId};
+use crate::signature_provider::{KeyshareId, RegisteredMpcTask, SignatureProvider};
+use crate::tracking;
+use borsh::{BorshDeserialize, BorshSerialize};
+use cait_sith::{FullSignature, KeygenOutput};
+use k256::{AffinePoint, Scalar, Secp256k1};
+use near_time::Clock;
+use std::sync::Arc;
+use tokio::sync::mpsc;
+
+#[derive(Clone)]
+pub struct EcdsaSecp256k1SignatureProvider {
+    config: Arc<ConfigFile>,
+    mpc_config: Arc<MpcConfig>,
+    client: Arc<MeshNetworkClient>,
+    triple_store: Arc<TripleStorage>,
+    presignature_store: Arc<PresignatureStorage>,
+    sign_request_store: Arc<SignRequestStorage>,
+    keygen_output: KeygenOutput<Secp256k1>,
+}
+
+impl EcdsaSecp256k1SignatureProvider {
+    pub fn new(
+        config: Arc<ConfigFile>,
+        mpc_config: Arc<MpcConfig>,
+        client: Arc<MeshNetworkClient>,
+        clock: Clock,
+        db: Arc<SecretDB>,
+        sign_request_store: Arc<SignRequestStorage>,
+        keygen_output: KeygenOutput<Secp256k1>,
+    ) -> anyhow::Result<Self> {
+        let active_participants_query = {
+            let network_client = client.clone();
+            Arc::new(move || network_client.all_alive_participant_ids())
+        };
+
+        let triple_store = Arc::new(TripleStorage::new(
+            clock.clone(),
+            db.clone(),
+            DBCol::Triple,
+            client.my_participant_id(),
+            |participants, pair| pair.is_subset_of_active_participants(participants),
+            active_participants_query.clone(),
+        )?);
+
+        let presignature_store = Arc::new(PresignatureStorage::new(
+            clock,
+            db.clone(),
+            DBCol::Presignature,
+            client.my_participant_id(),
+            |participants, presignature| {
+                presignature.is_subset_of_active_participants(participants)
+            },
+            active_participants_query,
+        )?);
+
+        Ok(Self {
+            config,
+            mpc_config,
+            client,
+            triple_store,
+            presignature_store,
+            sign_request_store,
+            keygen_output,
+        })
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, BorshSerialize, BorshDeserialize)]
+pub enum EcdsaSecp256k1TaskId {
+    KeyGeneration,
+    KeyResharing {
+        new_epoch: u64,
+    },
+    ManyTriples {
+        start: UniqueId,
+        count: u32,
+    },
+    Presignature {
+        id: UniqueId,
+        paired_triple_id: UniqueId,
+    },
+    Signature {
+        id: SignatureId,
+        presignature_id: UniqueId,
+    },
+}
+
+impl RegisteredMpcTask for EcdsaSecp256k1TaskId {
+    fn wrap(self) -> MpcTaskId {
+        MpcTaskId::EcdsaSecp256k1TaskId(self)
+    }
+}
+
+impl KeyshareId for KeygenOutput<Secp256k1> {
+    fn id() -> &'static str {
```

Comment:
I'd suggest still naming this keyshare_id() because on the KeygenOutput object we would be able to directly call .id() after importing this trait, and that would be confusing.